File: python/fib.py
#!/usr/bin/env python

import sys

def fib(n):
    last2 = 0
    last  = 1
    if n == 0 or n == 1:
        return n
    for i in range(2, n+1):
        f = last + last2
        last2 = last
        last = f
    return f
def main():
    ans = fib(int(sys.argv[1]))
    print "Fib=", ans

if __name__ == '__main__':
    main()

File: algo/fib.c
#include<stdio.h>

int mem[20];

int fib(int n)
{
	int k = 0;
	if (n < 2) return n;
	if (mem[n-1] == 0)
		mem[n-1] = fib(n-1);
	if (mem[n-2] == 0)
		mem[n-2] = fib(n-2);

	return mem[n-1] + mem[n-2];
}


int main(int argc, char *argv[])
{
	int i=0;
	for (i=0; i < 7; i++)
	{
		printf(" %d ", fib(i));
	}
}
